beater/config/java_attacher.go

```diff
@@ -0,0 +1,42 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package config
+
+// JavaAttacherConfig holds configuration information for running a java
+// attacher jarfile.
+type JavaAttacherConfig struct {
+	Enabled    bool `config:"enabled"`
+	Continuous bool `config:"continuous"`
+
+	IncludeAll bool     `config:"include_all"`
+	IncludePID []string `config:"include_pid"`
+
+	IncludeMain  []string `config:"include_main"`
+	IncludeVMArg []string `config:"include_vmarg"`
+	IncludeUser  []string `config:"include_user"`
+
+	ExcludeMain  []string `config:"exclude_main"`
+	ExcludeVMArg []string `config:"exclude_vmarg"`
+	ExcludeUser  []string `config:"exclude_user"`
```

Comment:
How did you define these rules, are they speced out somewhere?

Reply:
I looked at the docs online: https://www.elastic.co/guide/en/apm/agent/java/current/setup-attach-cli.html

I'm not 100% sure about what I included (or didn't include), just going off of the issue description that said 

>Mainly: java agent version, and include/exclude rules.

Since exactly what is happening still sounds a bit loosely defined, I decided to go with this for now and then continue from here with related issues